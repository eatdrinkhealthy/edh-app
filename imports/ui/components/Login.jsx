// @flow
import React, { Component } from "react";

class Login extends Component {
  props: {
    handleSubmit: (usernameEmail: string, password: string) => void,
  };

  state = {
    usernameEmail: "",
    loginPassword: "",
  };

  componentDidMount() {
    this.usernameEmailInput.focus();
  }

  usernameEmailInput: HTMLInputElement;

  resetForm = () => {
    this.setState({
      usernameEmail: "",
      loginPassword: "",
    });
    this.usernameEmailInput.focus();
  }

  handleInputChange = (event: Event) => {
    // NOTE: although you see this handleInputChange example in react docs, it is
    // getting a flow error. and may be pointing out why simulate change doesn't
    // work without setting node value first. (see notes in jest file).

    // $FlowFixMe
    const { value, name } = event.target;

    this.setState({
      [name]: value,
    });
  }

  onSubmit = (event: Event) => {
    event.preventDefault();

    this.resetForm();
    this.props.handleSubmit(this.state.usernameEmail, this.state.loginPassword);
  }

  render() {
    return (
      <div className="pt2 pl4 pr4 pb4">
        <form
          className="measure center"
          onSubmit={this.onSubmit}
        >
          <fieldset id="sign_in" className="ba b--transparent ph0 mh0">
            <legend className="f4 fw6 ph0 mh0">Login</legend>
            <div className="mt3">
              <label className="db fw6 lh-copy f6" htmlFor="usernameEmail">Username / Email</label>
              <input
                className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
                type="text"
                name="usernameEmail"
                id="usernameEmail"
                value={this.state.usernameEmail}
                onChange={this.handleInputChange}
                ref={(el: HTMLInputElement) => {
                  this.usernameEmailInput = el;
                }}
              />
            </div>
            <div className="mv3">
              <label className="db fw6 lh-copy f6" htmlFor="loginPassword">Password</label>
              <input
                className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
                type="password"
                name="loginPassword"
                id="loginPassword"
                value={this.state.loginPassword}
                onChange={this.handleInputChange}
              />
            </div>
          </fieldset>
          <div>
            <input
              className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
    );
  }
}

export default Login;
