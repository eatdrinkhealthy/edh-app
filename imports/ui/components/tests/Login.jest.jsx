// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import Login from "../Login";

describe("<Login />", function () {
  it("matches render snapshot", function () {
    function createNodeMock(element) {
      if (element.type === "input") {
        return {
          focus() {},
        };
      }
      return null;
    }
    const options = { createNodeMock };

    const tree = renderer.create(<Login handleSubmit={() => {}} />, options).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
