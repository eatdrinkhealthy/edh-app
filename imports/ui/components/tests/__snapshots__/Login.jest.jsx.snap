// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches render snapshot 1`] = `
<div
  className="pt2 pl4 pr4 pb4"
>
  <form
    className="measure center"
    onSubmit={[Function]}
  >
    <fieldset
      className="ba b--transparent ph0 mh0"
      id="sign_in"
    >
      <legend
        className="f4 fw6 ph0 mh0"
      >
        Login
      </legend>
      <div
        className="mt3"
      >
        <label
          className="db fw6 lh-copy f6"
          htmlFor="usernameEmail"
        >
          Username / Email
        </label>
        <input
          className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
          id="usernameEmail"
          name="usernameEmail"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mv3"
      >
        <label
          className="db fw6 lh-copy f6"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
          id="password"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </fieldset>
    <div>
      <input
        className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
        type="submit"
        value="Login"
      />
    </div>
  </form>
</div>
`;
